jupyterhub_traefik_proxy/traefik_utils.py

```diff
@@ -142,20 +136,55 @@ def atomic_writing(path):
             # already deleted by os.replace above
             pass
 
-
-def persist_static_conf(file, static_conf_dict):
-    with open(file, "w") as f:
-        toml.dump(static_conf_dict, f)
-
-
-def persist_routes(file, routes_dict):
-    with atomic_writing(file) as config_fd:
-        toml.dump(routes_dict, config_fd)
-
-
-def load_routes(file):
-    try:
-        with open(file, "r") as config_fd:
-            return toml.load(config_fd)
-    except:
-        raise
+class TraefikConfigFileHandler(object):
+    """Handles reading and writing Traefik config files. Can operate
+    on both toml and yaml files"""
+    def __init__(self, file_path):
+        file_ext = file_path.rsplit('.', 1)[-1]
+        if file_ext == 'yaml':
+            from ruamel.yaml import YAML
+            config_handler = YAML(typ="safe")
+        elif file_ext == 'toml':
+            import toml as config_handler
+        else:
+            raise TypeError("type should be either 'toml' or 'yaml'")
+
+        self.file_path = file_path
+        # Redefined to either yaml.dump or toml.dump
+        self._dump = config_handler.dump
+        #self._dumps = config_handler.dumps
+        # Redefined by __init__, to either yaml.load or toml.load
+        self._load = config_handler.load
+
+    def load(self):
+        """Depending on self.file_path, call either yaml.load or toml.load"""
+        with open(self.file_path, "r") as fd:
+            return self._load(fd)
+
+    def dump(self, data):
+        with open(self.file_path, "w") as f:
+            self._dump(data, f)
+
+    def atomic_dump(self, data):
+        """Save data to self.file_path after opening self.file_path with
+        :func:`atomic_writing`"""
+        with atomic_writing(self.file_path) as f:
+            self._dump(data, f)
+
+def persist_static_conf(file_path, static_conf_dict):
+    handler = TraefikConfigFileHandler(file_path)
+    handler.dump(static_conf_dict)
+
+def persist_dynamic_conf(file_path, routes_dict):
+    # FIXME: Only used by fileprovider, remove?
```

Comment:
Yes, just double checked and it isn't used any more. Was thinking about merging those methods into the `TraefikProxy` base class tbh. Either way, I have removed those standalone functions from `traefik_utils.py` in my working copy now. I did think that they might make nice static or class methods maybe, but it's been too long since I've written either...